istioctl/cmd/kubeuninject.go

```diff
@@ -0,0 +1,373 @@
+			} else {
+				volumes = append(volumes[:index])
+			}
+			//reset to 0
+			index = 0
+			l = len(volumes)
+		}
+		index++
+	}
+	return volumes
+}
+
+// handleAnnotations removes the injected sidecar.istio.io/status and sidecar.istio.io/rewriteAppHTTPProbers
+// it adds sidecar.istio.io/inject: false
+func handleAnnotations(annotations map[string]string) map[string]string {
+	if annotations == nil {
+		annotations = make(map[string]string)
+	}
+
+	for key := range annotations {
+		if key == annotationStatus {
+			delete(annotations, key)
+		}
+		if key == annotationRewriteAppHTTPProbers {
+			delete(annotations, key)
+		}
+	}
+	// sidecar.istio.io/inject: false to default the auto-injector in case it is present.
+	annotations[annotationPolicy] = "false"
+	return annotations
+}
+
+func extractObject(in runtime.Object) (interface{}, error) {
+	out := in.DeepCopyObject()
+
+	var metadata *metav1.ObjectMeta
+	var podSpec *corev1.PodSpec
+
+	// Handle Lists
+	if list, ok := out.(*corev1.List); ok {
+		result := list
+
+		for i, item := range list.Items {
+			obj, err := fromRawToObject(item.Raw)
+			if runtime.IsNotRegisteredError(err) {
+				continue
+			}
+			if err != nil {
+				return nil, err
+			}
+
+			r, err := extractObject(obj)
+			if err != nil {
+				return nil, err
+			}
+
+			re := runtime.RawExtension{}
+			re.Object = r.(runtime.Object)
+			result.Items[i] = re
+		}
+		return result, nil
+	}
+
+	// CronJobs have JobTemplates in them, instead of Templates, so we
+	// special case them.
+	switch v := out.(type) {
+	case *v2alpha1.CronJob:
+		job := v
+		metadata = &job.Spec.JobTemplate.ObjectMeta
+		podSpec = &job.Spec.JobTemplate.Spec.Template.Spec
+	case *corev1.Pod:
+		pod := v
+		metadata = &pod.ObjectMeta
+		podSpec = &pod.Spec
+	default:
+		// `in` is a pointer to an Object. Dereference it.
+		outValue := reflect.ValueOf(out).Elem()
+
+		templateValue := outValue.FieldByName("Spec").FieldByName("Template")
+		// `Template` is defined as a pointer in some older API
+		// definitions, e.g. ReplicationController
+		if templateValue.Kind() == reflect.Ptr {
+			templateValue = templateValue.Elem()
+		}
+		metadata = templateValue.FieldByName("ObjectMeta").Addr().Interface().(*metav1.ObjectMeta)
+		podSpec = templateValue.FieldByName("Spec").Addr().Interface().(*corev1.PodSpec)
+	}
+
+	podSpec.InitContainers = removeInjectedContainers(podSpec.InitContainers, initContainerName)
+	podSpec.Containers = removeInjectedContainers(podSpec.Containers, proxyContainerName)
+
+	podSpec.Volumes = removeInjectedVolumes(podSpec.Volumes)
+
+	metadata.Annotations = handleAnnotations(metadata.Annotations)
+
+	return out, nil
+}
+
+var (
+	uninjectInFilename  string
+	uninjectOutFilename string
+)
+
+func uninjectCommand() *cobra.Command {
+	uninjectCmd := &cobra.Command{
+		Use:   "kube-uninject",
+		Short: "Uninject Envoy sidecar from Kubernetes pod resources",
+		Long: `
+
+kube-uninject manually uninjects the Envoy sidecar from Kubernetes
```

Comment:
I realize this text is based on the `kube-inject` long description.  "manually" isn't the right word.  Can this text be simplified?